logic_mod/Blocks/Api/CpuMath.cs

```diff
@@ -114,9 +153,246 @@ public object Atan(VarCtx ctx, object[] x)
+                        // opposite direction to Newton.  Doesn't happen if x is close
+                        // enough to root.
+                        float adj = newton_step * fder2 / fder / 2;
+                        if (Math.Abs(adj) < 1)
+                            newton_step /= 1.0F - adj;
+                    }
+                    p = p0 - newton_step;
+                    if (WithinTol(p, p0, atol: tol, rtol: rtol))
+                        return _newton_result_select(full_output, p, itr + 1, converged: true);
+                    p0 = p;
+                }
+            }
+            else
+            {
+                // secant method
+                float p1, q0, q1;
+                if (x1Provided)
+                {
+                    if (x1 == x0)
+                        throw new Exception("x1 and x0 must be different");
+                    p1 = x1;
+                }
+                else
+                {
+                    float eps = 1e-4F;
+                    p1 = x0 * (1 + eps);
+                    p1 += (p1 >= 0 ? eps : -eps);
+                }
+                q0 = (float)Block.CallFunc(ctx, func, new object[] { p0 });
+                q1 = (float)Block.CallFunc(ctx, func, new object[] { p1 });
+                if (Math.Abs(q1) < Math.Abs(q0))
+                {
+                    float temp = q1;
+                    q1 = q0;
+                    q0 = temp;
+
+                    temp = p0;
+                    p0 = p1;
+                    p1 = temp;
+                }
+                for (; itr < maxiter; ++itr)
+                {
+                    if (q0 == q1)
+                    {
+                        p = (p1 + p0) / 2.0F;
+                        if (p1 != p0)
+                            return _newton_result_select(full_output, p, itr + 1, converged: false);
+                        else
+                            return _newton_result_select(full_output, p, itr + 1, converged: true);
+
+                    }
+                    else
+                    {
+                        // Secant Step
+                        if (Math.Abs(q1) > Math.Abs(q0))
+                            p = (-q0 / q1 * p1 + p0) / (1.0F - q0 / q1);
+                        else
+                            p = (-q1 / q0 * p0 + p1) / (1.0F - q1 / q0);
+                    }
+                    if (WithinTol(p, p1, atol: tol, rtol: rtol))
+                        return _newton_result_select(full_output, p, itr + 1, converged: true);
+
+                    p0 = p1;
+                    q0 = q1;
+                    p1 = p;
+                    q1 = (float)Block.CallFunc(ctx, func, new object[] { p1 });
+                }
+            }
+            return _newton_result_select(full_output, p, itr + 1, converged: false);
+        }
+
+        private object _newton_result_select(bool full_output, float p0,
+            int itr, bool converged)
```

Comment:
"int itr", passed outside native code, could break JS script further. Int is not supported by this JS, use "long"